octopusdeploy/schema_deployment_step.go

```diff
@@ -108,45 +91,59 @@ func flattenDeploymentSteps(deploymentSteps []*deployments.DeploymentStep) []map
 
 	var flattenedDeploymentSteps = make([]map[string]interface{}, len(deploymentSteps))
 	for key, deploymentStep := range deploymentSteps {
-		flattenedDeploymentSteps[key] = map[string]interface{}{}
-		flattenedDeploymentSteps[key]["condition"] = deploymentStep.Condition
-		flattenedDeploymentSteps[key]["id"] = deploymentStep.ID
-		flattenedDeploymentSteps[key]["name"] = deploymentStep.Name
-		flattenedDeploymentSteps[key]["package_requirement"] = deploymentStep.PackageRequirement
-		flattenedDeploymentSteps[key]["properties"] = flattenProperties(deploymentStep.Properties)
-		flattenedDeploymentSteps[key]["start_trigger"] = deploymentStep.StartTrigger
+		flattenedDeploymentStep := map[string]interface{}{}
+		flattenedDeploymentStep["condition"] = deploymentStep.Condition
+		flattenedDeploymentStep["id"] = deploymentStep.ID
+		flattenedDeploymentStep["name"] = deploymentStep.Name
+		flattenedDeploymentStep["package_requirement"] = deploymentStep.PackageRequirement
+		flattenedDeploymentStep["properties"] = flattenProperties(deploymentStep.Properties)
+		flattenedDeploymentStep["start_trigger"] = deploymentStep.StartTrigger
 
 		for propertyName, propertyValue := range deploymentStep.Properties {
 			switch propertyName {
 			case "Octopus.Action.TargetRoles":
-				flattenedDeploymentSteps[key]["target_roles"] = strings.Split(propertyValue.Value, ",")
+				flattenedDeploymentStep["target_roles"] = strings.Split(propertyValue.Value, ",")
 			case "Octopus.Action.MaxParallelism":
-				flattenedDeploymentSteps[key]["window_size"] = propertyValue.Value
+				flattenedDeploymentStep["window_size"] = propertyValue.Value
 			case "Octopus.Step.ConditionVariableExpression":
-				flattenedDeploymentSteps[key]["condition_expression"] = propertyValue.Value
+				flattenedDeploymentStep["condition_expression"] = propertyValue.Value
 			}
 		}
 
+		flatten_action_func := func(step_type_name string, i int, fp func(*deployments.DeploymentAction) map[string]interface{}) {
+
+			if _, ok := flattenedDeploymentStep[step_type_name]; !ok {
+				flattenedDeploymentStep[step_type_name] = make([]map[string]interface{}, 0)
+			}
+
+			action := fp(deploymentStep.Actions[i])
+			action["sort_order"] = i + 1
+			flattenedDeploymentStep[step_type_name] = append(flattenedDeploymentStep[step_type_name].([]map[string]interface{}), action)
+		}
+
 		for i := range deploymentStep.Actions {
 			switch deploymentStep.Actions[i].ActionType {
 			case "Octopus.KubernetesDeploySecret":
-				flattenedDeploymentSteps[key]["deploy_kubernetes_secret_action"] = []interface{}{flattenDeployKubernetesSecretAction(deploymentStep.Actions[i])}
```

Comment:
I think there was\is actually a bug here too, in that it would only ever store the last action per action type rather than the full array. Thats why the `terraform plan` when multiple actions were involved would always say changes are present, even if none were

Reply:
yes, I have come across that one too, nice